tests/fsharp/typecheck/sigs/neg04.bsl

```diff
@@ -62,10 +62,7 @@ neg04.fs(76,10,76,12): typecheck error FS0663: This type parameter has been used
 
 neg04.fs(76,19,76,26): typecheck error FS0698: Invalid constraint: the type used for the constraint is sealed, which means the constraint could only be satisfied by at most one solution
 
-neg04.fs(81,58,81,61): typecheck error FS0001: This expression was expected to have type
```

Comment:
Error is coming from this code:
```fsharp
let stringArrayDoesntSupport2DLookup (x : string) = x.[0,0]
```

The error message has changed. Was this the result of the removal of a rule in `typeSeqExpr`?

It has me a little worried. `0,0` by itself creates a tuple and there is only one overload of array's Item that takes an `int`; that's why the error message appeared as before makes sense.  The updated error message does not quite tell me what is wrong.

Reply:
This is just my best guess and I couldn't find any concrete evidence for this, but I think the error message has changed because of the way we rewrite tupled arguments in the type checker. If we change the parser to parse `.[1,2]` as `[SynIndexerArg.One(Const(1)); SynIndexerArg.One(Const(2))]` and tuple them inside the typechecker, this should go away.